snuba/datasets/schemas/join_schema.py

```diff
@@ -0,0 +1,90 @@
+from dataclasses import dataclass
+from enum import Enum
+from typing import Callable, Mapping, Optional, Sequence
+
+from snuba.clickhouse.columns import ColumnSet
+from snuba.datasets.schemas import Schema
+
+
+class JoinType(Enum):
+    INNER = "INNER"
+    LEFT = "LEFT"
+    RIGHT = "RIGHT"
+    FULL = "FULL"
+
+
+@dataclass(frozen=True)
+class JoinMapping():
+    left_alias: str
+    left_column: str
+    right_alias: str
+    right_column: str
+
+    def get_from_clause(self) -> str:
```

Comment:
I don't think `get_from_clause` is a good name for this. PostgreSQL calls this a `join_condition`, which I think would work, `join_expression` or `join_on_{clause,condition,expression}` or something along those lines I think would work too.